test/jdk/jdk/jfr/tool/TestMetadata.java

```diff
@@ -75,4 +87,65 @@ public static void main(String[] args) throws Throwable {
             lineNumber++;
         }
     }
-}
+
+    static void testEventTypeNum() throws Throwable {
+        OutputAnalyzer output = ExecuteHelper.jfr("metadata");
+        List<String> eventNames = new ArrayList<>();
+        List<String> lines = output.asLines();
+
+        for (String line : lines) {
+            if (line.startsWith("@Name(\"")) {
+                eventNames.add(line.substring(7, line.indexOf("\"", 7)));
+            }
+        }
+        List<EventType> eventTypes = FlightRecorder.getFlightRecorder().getEventTypes();
+        List<String> expectedNames = new ArrayList<>();
+        for (EventType eventType : eventTypes) {
+            expectedNames.add(eventType.getName());
+        }
+        Asserts.assertGTE(eventNames.size(), expectedNames.size());
+    }
+
+    static void testDeterministic() throws Throwable {
+        OutputAnalyzer output = ExecuteHelper.jfr("metadata", "--events", "CPULoad,GarbageCollection");
+        List<String> eventNames = new ArrayList<>();
+        List<String> lines = output.asLines();
+
+        for (String line : lines) {
+            if (line.startsWith("@Name(\"")) {
+                eventNames.add(line.substring(7, line.indexOf("\"", 7)));
```

Comment:
Should be easy to check if the event is correct, not just the number.

    if (line.contains("XXZQZYY.") {
      event1 = true;
    }
    if (line.contains("ZQZPP") {
      event2 = true;
    }
    
    Regarding events names, see following comment.